src/tarantool.client/Converters/ResponseHeaderConverter.cs

```diff
@@ -30,18 +31,31 @@ public void Write(ResponseHeader value, IMsgPackWriter writer)
 
         public ResponseHeader Read(IMsgPackReader reader)
         {
-            reader.ReadMapLength().ShouldBe(3u);
+            var length = reader.ReadMapLength();
+            if (length != 3u)
+            {
+                throw ExceptionHelper.InvalidMapLength(3u, length);
+            }
 
-            _keyConverter.Read(reader).ShouldBe(Key.Code);
+            ReadKey(reader, Key.Code);
```

Comment:
I will ask: will tarantool guarantee order of fields. But in general case it is not very to rely on order.